eclipse/src/saros/filesystem/EclipseFileImplV2.java

```diff
@@ -0,0 +1,149 @@
+package saros.filesystem;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.Charset;
+import org.apache.log4j.Logger;
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+
+/** Eclipse implementation of the Saros file interface. */
+// TODO rename to EclipseFile
+public class EclipseFileImplV2 extends EclipseResourceImplV2 implements IFile {
+  private static final Logger log = Logger.getLogger(EclipseFileImplV2.class);
+
+  /** @see EclipseResourceImplV2#EclipseResourceImplV2(EclipseReferencePointImpl, IPath) */
+  EclipseFileImplV2(EclipseReferencePointImpl referencePoint, IPath relativePath) {
+    super(referencePoint, relativePath);
+  }
+
+  @Override
+  public String getCharset() throws IOException {
+    try {
+      return getDelegate().getCharset();
+
+    } catch (CoreException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public void setCharset(String charset) throws IOException {
+    if (charset == null) {
+      return;
+    }
+
+    // Check whether charset is valid and supported
+    Charset.forName(charset);
+
+    try {
+      updateFileEncoding(charset);
+
+    } catch (CoreException | OperationCanceledException e) {
```

Comment:
According to the javadoc of `setEncoding()`, the exception can be thrown even if no progress monitor is passed (see [`IFile.setEncoding()`](https://help.eclipse.org/2019-12/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fresources%2FIFile.html)).

It is currently already re-thrown as an `IOException`. Additional logging would be possible, but it would break with the rest of the behavior of the methods, which generally just throw the exceptions and let the caller deal with it.